cli/src/cmd/forge/verify.rs

```diff
@@ -235,6 +306,93 @@ To skip this solc dry, pass `--force`.
     }
+
+    fn flattened_source(
+        &self,
+        project: &Project,
+        target: &Path,
+        version: &Version,
+    ) -> eyre::Result<(String, String, CodeFormat)> {
+        let bch = project
+            .solc_config
+            .settings
+            .metadata
+            .as_ref()
+            .and_then(|m| m.bytecode_hash)
+            .unwrap_or_default();
+
+        eyre::ensure!(
+            bch == BytecodeHash::Ipfs,
+            "When using flattened source, bytecodeHash must be set to ipfs. BytecodeHash is currently: {}. Hint: Set the bytecodeHash key in your foundry.toml :)",
+            bch,
+        );
+
+        let source = project.flatten(target).wrap_err("Failed to flatten contract")?;
+
+        if !self.force {
+            // solc dry run of flattened code
+            self.check_flattened(source.clone(), version).map_err(|err| {
+                eyre::eyre!(
+                    "Failed to compile the flattened code locally: `{}`\
+    To skip this solc dry, have a look at the  `--force` flag of this command.",
+                    err
+                )
+            })?;
+        }
+
+        let name = self.contract.name.clone();
+        Ok((source, name, CodeFormat::SingleFile))
+    }
+
+    fn standard_json_source(
+        &self,
+        project: &Project,
+        target: &Path,
+        version: &Version,
+    ) -> eyre::Result<(String, String, CodeFormat)> {
+        let input = project
+            .standard_json_input(target)
+            .wrap_err("Failed to get standard json input")?
+            .normalize_evm_version(version);
+
+        let source =
+            serde_json::to_string(&input).wrap_err("Failed to parse standard json input")?;
+        let name = format!(
+            "{}:{}",
+            target.strip_prefix(project.root()).unwrap_or(target).display(),
+            self.contract.name.clone()
+        );
+        Ok((source, name, CodeFormat::StandardJsonInput))
+    }
+}
+
+/// Strips [BuildMetadata] from the [Version]
+///
+/// **Note:** this is only for local compilation as a dry run, therefore this will return a
+/// sanitized variant of the specific version so that it can be installed. This is merely
+/// intended to ensure the flattened code can be compiled without errors.
+fn strip_build_meta(version: Version) -> Version {
+    if version.build != BuildMetadata::EMPTY {
+        Version::new(version.major, version.minor, version.patch)
+    } else {
+        version
+    }
+}
+
+/// Given any solc [Version] return a [Version] with build metadata
+///
+/// # Example
+/// ```
+/// let version = Version::new(1, 2, 3);
+/// let version = ensure_solc_build_metadata(version).await?;
+/// assert_ne!(version.build, BuildMetadata::EMPTY);
+/// ```
+async fn ensure_solc_build_metadata(version: Version) -> eyre::Result<Version> {
+    if version.build != BuildMetadata::EMPTY {
+        Ok(version)
+    } else {
+        Ok(lookup_compiler_version(&version).await?)
```

Comment:
Instead of always doing a network request, is there a way to pull this in `ethers-etherscan` via a `build.rs` or something? cc @mattsse

Reply:
from what i've seen, the solc commit metadata that gets written into cache is taken from running `solc --version` [here](https://github.com/gakonst/ethers-rs/blob/f5efbbb86a4f93e0db1ae897de18b4e032ad49d2/ethers-solc/src/compile/mod.rs#L500). and most likely we will detect it from cache now anyway, this network request is more of an edge case handler when the user passes in explicit compiler version with the flag

Reply:
yeh metadata is part of the `--version`, 

a request as fallback is probably negligible for create+verify anyways